Source: William Orr <will@worrbase.com>
Upstream: https://github.com/gugod/App-perlbrew/pull/289
Reason: Fixes uninitialized warnings
---
 lib/App/perlbrew.pm | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/App/perlbrew.pm b/lib/App/perlbrew.pm
index 48f1888..2064ec1 100644
--- a/lib/App/perlbrew.pm
+++ b/lib/App/perlbrew.pm
@@ -1359,11 +1359,13 @@ WARNINGONMAC
 
     my $command = "env ";
     while (my ($k, $v) = each(%env)) {
+        no warnings "uninitialized";
         $command .= "$k=\"$v\" ";
     }
     $command .= " $shell $shell_opt";
 
-    print "\nA sub-shell is launched with $name as the activated perl. Run 'exit' to finish it.\n\n";
+    my $pretty_name = defined($name) ? $name : "the default perl";
+    print "\nA sub-shell is launched with $pretty_name as the activated perl. Run 'exit' to finish it.\n\n";
     exec($command);
 }
 
-- 
1.8.1.3

